File: decompiled_235_src/sources/me/magnum/melonds/MelonDSiNand.java
package me.magnum.melonds;

import java.util.ArrayList;
import me.magnum.melonds.domain.model.DSiWareTitle;
import me.magnum.melonds.domain.model.EmulatorConfiguration;
/* compiled from: r8-map-id-0dd4387375ec438f6a4c042bfe6cb4a3262713dcd4829883cccf19288831b1df */
/* loaded from: classes.dex */
public final class MelonDSiNand {
    public static final MelonDSiNand a = new Object();

    public final native void closeNand();

    public final native void deleteTitle(int i);

    public final native boolean exportTitleExecutable(int i, String str);

    public final native boolean exportTitleFile(int i, int i2, String str);

    public final native int importTitle(String str, byte[] bArr);

    public final native boolean importTitleFile(int i, int i2, String str);

    public final native ArrayList<DSiWareTitle> listTitles();

    public final native int openNand(EmulatorConfiguration emulatorConfiguration);

    public final native boolean repairTitleSaves(int i);
}

File: app/src/main/java/me/magnum/melonds/MelonDSiNand.kt
package me.magnum.melonds

import me.magnum.melonds.domain.model.DSiWareTitle
import me.magnum.melonds.domain.model.EmulatorConfiguration

object MelonDSiNand {
    external fun openNand(emulatorConfiguration: EmulatorConfiguration): Int
    external fun listTitles(): ArrayList<DSiWareTitle>
    external fun importTitle(titleUri: String, tmdMetadata: ByteArray): Int
    external fun deleteTitle(titleId: Int)
    external fun repairTitleSaves(titleId: Int): Boolean
    external fun ensureTitleSaveStructure(titleId: Int, romHeaderBytes: ByteArray, tmdMetadata: ByteArray?): Boolean
    external fun ensureSystemFontTable(fontData: ByteArray): Boolean
    external fun exportTitleExecutable(titleId: Int, outputPath: String): Boolean
    external fun importTitleFile(titleId: Int, fileType: Int, fileUri: String): Boolean
    external fun exportTitleFile(titleId: Int, fileType: Int, fileUri: String): Boolean
    external fun closeNand()
}
